File: src/algorithms/insertionSort.ts
export default function insertionSort<T>(array: T[]): T[] {
	for (let i = 0; i < array.length; i++) {
		let j = i;

		while (j > 0 && array[j] < array[j - 1]) {
			[array[j - 1], array[j]] = [array[j], array[j - 1]];
			j--;
		}
	}

	return array;
}

File: tests/algorithms/sort/insertion_sort.cpp
#include <array>
#include <print>

import insertion_sort;

auto main() -> int {
	auto array = std::array{4, 2, 1, 7, 6};

	std::print("original: ");
	for (auto i : array) {
		std::print("{} ", i);
	}
	std::println("");

	insertion_sort(array);

	std::print("sorted: ");
	for (auto i : array) {
		std::print("{} ", i);
	}
	std::println("");
}
